src/sage/categories/fields.py

```diff
@@ -652,7 +657,7 @@ def gcd(self,other):
                 from sage.rings.integer_ring import ZZ
                 try:
                     return P(ZZ(self).gcd(ZZ(other)))
-                except TypeError:
+                except (TypeError, ValueError):
```

Comment:
I think it is good to widen what we catch. Yes, we probably should be more consistent, but that isn't a reason for this to fail. Do you have an example where it was breaking before this change?

Reply:
@mantepse Do you have an answer to this question? This is the only question that remains from me.

Reply:
I'm sorry, I overlooked this comment.  Yes, without this change we have, for example
```
sage: QQbar(0).gcd(QQbar.zeta(3))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[3], line 1
----> 1 QQbar(Integer(0)).gcd(QQbar.zeta(Integer(3)))

File ~/sage-trac/src/sage/structure/element.pyx:4742, in sage.structure.element.coerce_binop.new_method()
   4740 def new_method(self, other, *args, **kwargs):
   4741     if have_same_parent(self, other):
-> 4742         return method(self, other, *args, **kwargs)
   4743     else:
   4744         a, b = coercion_model.canonical_coercion(self, other)

File ~/sage-trac/src/sage/categories/fields.py:691, in Fields.ElementMethods.gcd(self, other)
    689 from sage.rings.integer_ring import ZZ
    690 try:
--> 691     return P(ZZ(self).gcd(ZZ(other)))
    692 except TypeError:
    693     pass

File ~/sage-trac/src/sage/structure/parent.pyx:901, in sage.structure.parent.Parent.__call__()
    899 if mor is not None:
    900     if no_extra_args:
--> 901         return mor._call_(x)
    902     else:
    903         return mor._call_with_args(x, args, kwds)

File ~/sage-trac/src/sage/structure/coerce_maps.pyx:293, in sage.structure.coerce_maps.NamedConvertMap._call_()
    291     raise TypeError(f"cannot coerce {x} to {C}")
    292 cdef Map m
--> 293 cdef Element e = method(C)
    294 if e is None:
    295     raise RuntimeError("BUG in coercion model: {} method of {} returned None".format(self.method_name, type(x)))

File ~/sage-trac/src/sage/rings/qqbar.py:5288, in AlgebraicNumber._integer_(self, ZZ)
   5261 def _integer_(self, ZZ=None):
   5262     """
   5263     Return ``self`` as an Integer.
   5264 
   (...)
   5286         16
   5287     """
-> 5288     return AA(self)._integer_(ZZ)

File ~/sage-trac/src/sage/structure/parent.pyx:901, in sage.structure.parent.Parent.__call__()
    899 if mor is not None:
    900     if no_extra_args:
--> 901         return mor._call_(x)
    902     else:
    903         return mor._call_with_args(x, args, kwds)

File ~/sage-trac/src/sage/structure/coerce_maps.pyx:164, in sage.structure.coerce_maps.DefaultConvertMap_unique._call_()
    162             print(type(C), C)
    163             print(type(C._element_constructor), C._element_constructor)
--> 164         raise
    165 
    166 cpdef Element _call_with_args(self, x, args=(), kwds={}):

File ~/sage-trac/src/sage/structure/coerce_maps.pyx:159, in sage.structure.coerce_maps.DefaultConvertMap_unique._call_()
    157 cdef Parent C = self._codomain
    158 try:
--> 159     return C._element_constructor(x)
    160 except Exception:
    161     if print_warnings:

File ~/sage-trac/src/sage/rings/qqbar.py:1160, in AlgebraicRealField._element_constructor_(self, x)
   1158         return x.real()
   1159     else:
-> 1160         raise ValueError("Cannot coerce algebraic number with nonzero imaginary part to algebraic real")
   1161 elif hasattr(x, '_algebraic_'):
   1162     return x._algebraic_(AA)

ValueError: Cannot coerce algebraic number with nonzero imaginary part to algebraic real
```
because `ZZ(QQbar.zeta(3))` raises a `ValueError` and not a `TypeError`.

Reply:
Okay, thanks.